lib/spack/spack/modules/common.py

```diff
@@ -763,6 +769,29 @@ def has_manpath_modifications(self):
         else:
             return False
 
+    @tengine.context_property
+    def conflicts(self):
+        """List of conflicts for the module file."""
+        fmts = []
+        projection = proj.get_projection(self.conf.projections, self.spec)
+        f = string.Formatter()
+        for item in self.conf.conflicts:
+            if len([x for x in f.parse(item)]) > 1:
+                for naming_dir, conflict_dir in zip(projection.split("/"), item.split("/")):
+                    if naming_dir != conflict_dir:
+                        message = "conflict scheme does not match naming "
+                        message += "scheme [{spec}]\n\n"
+                        message += 'naming scheme   : "{nformat}"\n'
+                        message += 'conflict scheme : "{cformat}"\n\n'
+                        message += "** You may want to check your "
+                        message += "`modules.yaml` configuration file **\n"
+                        tty.error(message.format(spec=self.spec, nformat=projection, cformat=item))
+                        raise SystemExit("Module generation aborted.")
```

Comment:
This can be extracted as a separate function, with a more meaningful name. Maybe:
```python
self._verify_conflict_naming_consistency_or_raise(item)
```
?